src/WebJobs.Script.WebHost/ProxyFunctionExecutor.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http.Dependencies;
+using Microsoft.Azure.AppService.Proxy.Client.Contract;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Controllers;
+using Microsoft.Azure.WebJobs.Script.WebHost.WebHooks;
+
+namespace Microsoft.Azure.WebJobs.Script.Host
+{
+    public class ProxyFunctionExecutor : IFuncExecutor
+    {
+        private readonly WebScriptHostManager _scriptHostManager;
+        private readonly IDependencyResolver _dependencyResolver;
+        private WebHookReceiverManager _webHookReceiverManager;
+
+        internal ProxyFunctionExecutor(WebScriptHostManager scriptHostManager, IDependencyResolver dependencyResolver, WebHookReceiverManager webHookReceiverManager)
+        {
+            _scriptHostManager = scriptHostManager;
+            _webHookReceiverManager = webHookReceiverManager;
+            _dependencyResolver = dependencyResolver;
+        }
+
+        public async Task ExecuteFuncAsync(string funcName, Dictionary<string, object> arguments, CancellationToken cancellationToken)
+        {
+            HttpRequestMessage request = arguments[ScriptConstants.AzureFunctionsHttpRequestKey] as HttpRequestMessage;
+            var function = _scriptHostManager.GetHttpFunctionOrNull(request);
+
+            var response = await FunctionRequestInvoker.PreValidateRequestAsync(request, function, _dependencyResolver);
+
+            if (response != null)
+            {
+                request.Properties[ScriptConstants.AzureFunctionsHttpResponseKey] = response;
+                return;
+            }
+
+            Func<HttpRequestMessage, CancellationToken, Task<HttpResponseMessage>> processRequestHandler = async (req, ct) =>
+            {
+                return await FunctionRequestInvoker.ProcessRequestAsync(req, function, ct, _scriptHostManager, _webHookReceiverManager);
+            };
+
+            var resp = await _scriptHostManager.HttpRequestManager.ProcessRequestAsync(request, processRequestHandler, cancellationToken);
```

Comment:
Should you be pumping this into the HttpRequestManager again here? If the request has already been initially processed by the FunctionsController, it has already been through the manager throttles - wouldn't this mean it would happen twice?

Reply:
Yes and this is expected. the incoming request is for a proxy endpoint and in proxy the request properties might have been changed with requestoverrides so once all the overrides have been processed and we have determined this is a local function, we will need to pump it thru again. the main purpose of this PR is not to send the full request to FEs and such and hit sandbox limitations, ... when the call is to a local function.